src/tags/object/Video/Rectangle.js

```diff
@@ -21,46 +22,56 @@ const RectanglePure = ({ reg, frame, workingArea, ...rest }) => {
 
   if (!box) return null;
 
-  const { realWidth: waWidth, realHeight: waHeight } = workingArea;
+  const { realWidth: waWidth, realHeight: waHeight, scale: waScale } = workingArea;
 
-  const newBox = {
+  const newBox = useMemo(()=>({
     x: box.x * waWidth / 100,
     y: box.y * waHeight / 100,
     width: box.width * waWidth / 100,
     height: box.height * waHeight / 100,
     rotation: box.rotation,
-  };
+  }), [box, waWidth, waHeight]);
 
   return reg.isInLifespan(frame) ? (
-    <Rect
-      {...newBox}
-      draggable
-      fill={style.fillColor}
-      stroke={style.strokeColor}
-      strokeScaleEnabled={false}
-      opacity={reg.hidden ? 0 : 1}
-      onTransformEnd={e => {
-        const node = e.target;
-        const scaleX = node.scaleX();
-        const scaleY = node.scaleY();
-        // set minimal value
-        const w = Math.max(3, node.width() * scaleX);
-        const h = Math.max(3, node.height() * scaleY);
+    <Group>
+      <LabelOnVideoBbox
+        reg={reg}
+        box={newBox}
+        scale={waScale}
+        color={style.strokeColor}
+        strokeWidth={style.strokeWidth}
+        adjacent
+      />
+      <Rect
+        {...newBox}
+        draggable
+        fill={style.fillColor}
+        stroke={style.strokeColor}
+        strokeScaleEnabled={false}
+        opacity={reg.hidden ? 0 : 1}
+        onTransformEnd={e => {
+          const node = e.target;
+          const scaleX = node.scaleX();
+          const scaleY = node.scaleY();
+          // set minimal value
+          const w = Math.max(3, node.width() * scaleX);
+          const h = Math.max(3, node.height() * scaleY);
```

Comment:
can we use a const variable to define what 3 is - the 3 is just in there and I dont understand what it represents (aside from being a safe value in case node dimensions are too small)

Reply:
Sure! I think I have found the right constant. And yeah, it may be too small, but it was done not by me and it is the same for the ImageView regions. So I think for now it is the right value.